src/makeRobot.js

```diff
@@ -39,6 +39,96 @@
  */
 function makeRobot(name, wheels, version) {
   // write code here
+
+  const robot = {
+
+    name,
+    wheels,
+    version,
+    coords: {
+      x: 0,
+      y: 0,
+    },
+
+    set info(info) {
+      this.name = info[0];
+      this.wheels = info[1];
+      this.version = info[2];
+    },
+
+    get info() {
+      return `name: ${this.name}, \
+chip version: ${this.version}, \
+wheels: ${this.wheels}`;
+    },
+
+    get location() {
+      return `${this.name}: x=${this.coords.x}, y=${this.coords.y}`;
+    },
+
+    goForward(value) {
+      if (value < 0) {
+        return this;
+      }
+
+      if (typeof value === 'number' && value > 1) {
+        this.coords.y += value;
+      } else {
+        this.coords.y++;
+      }
+
+      return this;
+    },
+
+    goBack(value = 0) {
+      if (value > 1) {
+        this.coords.y -= value;
+      }
+
+      if (value === 1 || value === 0) {
+        this.coords.y--;
+      }
+
+      return this;
+    },
```

Comment:
try to do this with only 1 if statement. Hint: set 1 as a default value initially. Refactor all of the functions in a similar way, not only goBack

Reply:
this should still be fixed for all of the functions (goBack, goForward....)

Reply:
> try to do this with only 1 if statement. Hint: set 1 as a default value initially. Refactor all of the functions in a similar way, not only goBack

Start with this
```suggestion
     goBack(value = 1) {
     //rest of the code
     }
```
After this you will need only one if statement to check if the value is greater than 0